src/parsed_source.rs

```diff
@@ -77,61 +81,52 @@ pub(crate) struct ParsedSourceInner {
 ///
 /// Note: This struct is cheap to clone.
 #[derive(Clone)]
-pub struct ParsedSource {
-  pub(crate) inner: Arc<ParsedSourceInner>,
-}
+pub struct ParsedSource(pub(crate) Arc<ParsedSourceInner>);
 
 impl ParsedSource {
-  #[allow(clippy::too_many_arguments)]
-  pub(crate) fn new(
-    specifier: ModuleSpecifier,
-    media_type: MediaType,
-    text_info: SourceTextInfo,
-    comments: MultiThreadedComments,
-    program: Arc<Program>,
-    tokens: Option<Arc<Vec<TokenAndSpan>>>,
-    globals: Globals,
-    syntax_contexts: Option<SyntaxContexts>,
-    diagnostics: Vec<ParseDiagnostic>,
-  ) -> Self {
-    ParsedSource {
-      inner: Arc::new(ParsedSourceInner {
-        specifier,
-        media_type,
-        text_info,
-        comments,
-        program,
-        tokens,
-        globals,
-        syntax_contexts,
-        diagnostics,
-      }),
-    }
-  }
-
   /// Gets the module specifier of the module.
   pub fn specifier(&self) -> &ModuleSpecifier {
-    &self.inner.specifier
+    &self.0.specifier
   }
 
   /// Gets the media type of the module.
   pub fn media_type(&self) -> MediaType {
-    self.inner.media_type
+    self.0.media_type
   }
 
   /// Gets the text content of the module.
-  pub fn text_info(&self) -> &SourceTextInfo {
-    &self.inner.text_info
+  pub fn text(&self) -> &Arc<str> {
+    &self.0.text
+  }
+
+  /// Gets an object with pre-computed positions for lines and indexes of
+  /// multi-byte chars.
+  ///
+  /// Note: Prefer using `.text()` over this if able because this is lazily
+  /// created.
+  pub fn text_info_lazy(&self) -> &SourceTextInfo {
```

Comment:
I gave this a different name than before in order to discourage using it accidentally.